Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -13598,6 +13603,41 @@ - (void)_closeCurrentTypingCommand
     _page->closeCurrentTypingCommand();
 }
 
+- (UIView *)_selectionContainerViewAboveText
+{
+    // FIXME: Once -selectionContainerView (or an equivalent) is available as API, we should adopt that and
+    // make this invoke RELEASE_ASSERT_ASYNC_TEXT_INTERACTIONS_DISABLED() instead.
+    return self._selectionContainerViewInternal;
+}
+
+- (UIView *)selectionContainerView
+{
+    return self._selectionContainerViewInternal;
+}
+
+- (UIView *)_selectionContainerViewInternal
+{
+    if (!self.selectionHonorsOverflowScrolling)
+        return self;
+
+    if (!_page->editorState().hasVisualData())
+        return self;
+
+    auto scrollingNodeID = _page->editorState().visualData->enclosingScrollingNodeID;
+    if (!scrollingNodeID)
+        return self;
+
+    auto* coordinator = downcast<WebKit::RemoteScrollingCoordinatorProxyIOS>(_page->scrollingCoordinatorProxy());
```

Comment:
should this be some sort of smart pointer instead?

Reply:
Sort of — it implements `CanMakeWeakPtr` so we can at least hold a weak reference, though I think we'd ideally make this a `CheckedPtr` since the coordinator should never be destroyed as a result of this logic.

I'll make this a weak pointer for now.